src/components/data-table/ha-data-table.ts

```diff
@@ -369,20 +372,25 @@ export class HaDataTable extends LitElement {
     `;
   }
 
-  private async _filterData() {
+  private async _sortFilterData() {
     const startTime = new Date().getTime();
     this.curRequest++;
     const curRequest = this.curRequest;
 
-    const filterProm = filterSortData(
-      this.data,
-      this._sortColumns,
-      this._filter,
-      this._sortDirection,
-      this._sortColumn
-    );
+    const filteredData: Promise<DataTableRowData[]> = this._filter
+      ? this._memFilterData(this.data, this._sortColumns, this.filter)
+      : new Promise((resolve) => resolve(this.data));
```

Comment:
```suggestion
      : Promise.resolve(this.data);
```

However, why not do this and skip 1 await if there is no filter:

```ts
let filteredData = this.data;
if (this._filter) {
    filteredData = await …
}